src/lib.rs

```diff
@@ -169,6 +109,43 @@ impl Default for MaybeInitializedCaptureIndex {
 
 const ALL_NODES_QUERY: &str = "(_) @node";
 
+struct CachedQueries(HashMap<SupportedLanguageName, OnceLock<Option<Arc<Query>>>>);
+
+impl CachedQueries {
+    fn get_and_cache_query_for_language(
+        &self,
+        query_source: &str,
+        language: &dyn SupportedLanguage,
+    ) -> Option<Arc<Query>> {
+        self.0
+            .get(&language.name())
+            .unwrap()
+            .get_or_init(|| maybe_get_query(query_source, language.language()).map(Arc::new))
+            .clone()
+    }
+
+    fn error_if_no_successful_query_parsing(&self) {
+        if !self
+            .0
+            .values()
+            .any(|query| query.get().and_then(|option| option.as_ref()).is_some())
```

Comment:
In case it is interesting:

I first tried to write this as `query.get().flatten().is_some()`

But that wasn't allowed because technically `.get()` gives back an `Option<&Option<...>>` not an `Option<Option<...>>`

So then I thought "ok either I can `.clone()` the inner `Option` (so `.cloned()` at the top level), but `Arc`-cloning isn't completely "free" (so probably being "naive about optimization guy"), or `.as_ref()` the inner `Option` to go `Option<&Option<...>>` -> `Option<Option<&...>>`" (and then be able to use `.flatten()`)

So I did the latter and then Clippy suggested that `.map(|option| option.as_ref()).flatten()` could be written as `.and_then(|option| option.as_ref())` instead

(conceptually the outer `Option` here is "was the `OnceLock` for this language ever initialized" and the inner `Option` is "given that it was in fact initialized, did the query fail to parse for that language")